xbmc/addons/AddonInstaller.cpp

```diff
@@ -116,12 +116,10 @@ void CAddonInstaller::GetInstallList(VECADDONS &addons) const
   }
   lock.unlock();
 
-  CAddonDatabase database;
-  database.Open();
   for (std::vector<std::string>::iterator it = addonIDs.begin(); it != addonIDs.end(); ++it)
   {
     AddonPtr addon;
-    if (database.GetAddon(*it, addon))
+    if (CServiceBroker::GetAddonMgr().FindInstallableById(*it, addon))
```

Comment:
you could save some cpu cycles if you obtain the value of `CServiceBroker::GetAddonMgr()` once outside the loop, not on every iteration.

Reply:
done